pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -26,7 +26,7 @@ import (
 
 	"knative.dev/pkg/apis"
 	"knative.dev/serving/pkg/apis/autoscaling"
-	autoscalerconfig "knative.dev/serving/pkg/autoscaler/config"
+	"knative.dev/serving/pkg/autoscaler/config/sharedconfig"
```

Comment:
this may well be a terrible idea, but what if we called this package `autoscalerconfig`? Then we'd maybe only have to change the imports - since we're always importing `autoscaler/config` as this anyway - and the diff would be much smaller (also it seems like maybe a better name for this anyway). Unless there's a place we import both `autoscaler/config/sharedconfig` and `autoscaler/config`, but I don't think there can be because it'd be circular then anyway?

Reply:
There are a few places that we need to import both, such as `test/e2e/e2e.go`. It's mostly `ConfigName` and `NewConfigFromMap(...)` that we are importing from `autoscaler/config`. I tried to move them to `sharedconfig` but then we were going back to the circular dependency we had to begin with.

Reply:
can't you just import them with different names in those couple places? I guess at that point it becomes a bit icky anyway

Reply:
Ah I see. So instead of `autoscalerconfig "knative.dev/serving/pkg/autoscaler/config"` we call it `asconfig "knative.dev/serving/pkg/autoscaler/config"`, and rename `autoscaler/config/sharedconfig` to `autoscaler/config/autoscalerconfig` ?

Reply:
yeah, dont honestly know if that's better or worse if we end up having to import both `asconfig` and `autoscalerconfig` in the same file tho. Not that `autoscalerconfig` + `sharedconfig` is a ton better so 🤷

Reply:
certainly makes the diff look nicer this way 🤷‍♂️👍️.